### PR TITLE
Add composite propagators

### PR DESCRIPTION
**Pull request by plantfansam**
The OpenTelemetry spec includes [Composite Propagators](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/context/api-propagators.md#composite-propagator), which allow you to run several propagators in sequence. This work adds composite propagators. Along the way, it makes propagators configurable, so that users can decide how their context gets propagated (the spec [requires support for b3, jaeger, and w3c baggage propagators](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/context/api-propagators.md#propagators-distribution)).

In order to add composite propagators, I did a few things:

1. relocated `trace_context.lua` to `text_map/trace_context_propagator.lua`.
1. deleted `carrier.lua`, since the code is not using it anymore
1. updated the method signature of inject and extract to include optional getter/setter arguments (see [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/context/api-propagators.md#getter-argument)).
1. added modules for composite propagator, getter, and setter.
1. updated nginx.conf in the `examples` directory.

I also added a test harness for [`busted`](https://olivinelabs.com/busted/) tests, along with a `make` command to run them.

These changes will require existing users to modify their code. If that is not acceptable, we can find a solution 😄. 

I'm new to Lua, so I'm sure there are plenty of changes required, but I wanted to get some 👀 .

### CODE REVIEW

**Comment by yangxikun** [comment]
Please rebase main branch.

**Comment by yangxikun** [comment]
@plantfansam Please check the review.

**Comment by yangxikun** [comment]
@plantfansam You cannot see this review?

**Comment by plantfansam** [comment]
@yangxikun I cannot see that review. The comments are shown as pending in the screenshot you posted. I think you need to click "submit review," like this:

Screen Shot 2022-06-28 at 8 41 15 AM

**Review by yangxikun — CHANGES_REQUESTED** [review]

File: lib/opentelemetry/trace/propagation/text_map/composite_propagator.lua [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    return setmetatable({ propagators = propagators }, mt)
+end
+
+--------------------------------------------------------------------------------
+-- Uses the propagators to inject context into carrier in sequence.
+--
+-- @param context      context module
+-- @param carrier      carrier (e.g. ngx.req)
+--------------------------------------------------------------------------------
+function _M:composite_inject(context, carrier)
```
[the remark this answers is not in the document]
Comment (plantfansam) [reply lib/opentelemetry/trace/propagation/text_map/composite_propagator.lua]: 👍 22b1afdaaf7f875acd0752d3fc4908210d388f1b

File: lib/opentelemetry/trace/id_generator.lua [hunk 2]
```diff
@@ -6,7 +6,7 @@ local random = math.random
 
 local _M = {}
 
-math.randomseed(ngx.time() + ngx.worker.pid())
+math.randomseed(os.time(os.date("!*t")))
```
[the remark this answers is not in the document]
Comment (plantfansam) [reply lib/opentelemetry/trace/id_generator.lua]: ✅  cf0b82cb34b6c85364b4a1f1fc6615c7cc4936ba

File: lib/opentelemetry/trace/propagation/text_map/getter.lua [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+local mt = {
+    __index = _M
+}
+
+function _M.new()
+    return setmetatable({}, mt)
+end
+
+------------------------------------------------------------------
+-- Add tracing information to nginx request as headers
```
Comment (yangxikun) [inline lib/opentelemetry/trace/propagation/text_map/getter.lua:13]: wrong doc

**Review by yangxikun — APPROVED** [review]

**Pull request merged**